deepchem/data/datasets.py

```diff
@@ -374,7 +380,8 @@ def itersamples(self):
     """
     raise NotImplementedError()
 
-  def transform(self, fn, **args):
+  def transform(self, fn: Callable[[np.ndarray, np.ndarray, np.ndarray], Tuple[
```

Comment:
Could we introduce an auxiliary type here? Say

```
TransformerBatch = Tuple[np.ndarray, np.ndarray, np.ndarray]
```
Then we write something like
```
Callable[TransformerBatch, TransformerBatch]
```

Reply:
The argument to `fn` isn't a tuple, it's just three separate arrays.  I wonder if we should change this method to take a Transformer object instead?  I'm investigating this as part of #1990.

Reply:
Ah that's interesting. Only transformers use the `Dataset.transform` method in any case so might be worth tying them together if it helps clean things up